packages/astro/src/integrations/astroFeaturesValidation.ts

```diff
@@ -88,6 +90,14 @@ export function validateSupportedFeatures(
 		}
 	}
 
+	validationResult.env = validateSupportKind(
+		env,
+		adapterName,
+		logger,
+		'astro:env',
+		() => config?.experimental?.env?.schema !== undefined
+	);
```

Comment:
The validation should be triggered only `config?.experimental?.env` isn't `undefined`. Otherwise, all adapters will start logging the warning even though a user didn't enable the feature.